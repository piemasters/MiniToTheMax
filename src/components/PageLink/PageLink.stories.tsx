import type { Meta, StoryObj } from '@storybook/react';

import { PageLink } from './PageLink';

const meta: Meta<typeof PageLink> = {
  component: PageLink,
  argTypes: {
    type: {
      name: 'type',
      description: 'The animation type',
      defaultValue: 'paintDrip',
      options: ['paintDrip', 'fade', 'swipe', 'cover'],
      control: {
        type: 'select',
      },
    },
    to: {
      name: 'to',
      description: 'The internal link address',
      defaultValue: '/',
      control: { type: 'text' },
    },
    direction: {
      name: 'direction',
      description: 'Directional of transitions',
      defaultValue: 'up',
      options: ['left', 'right', 'up', 'down'],
      control: { type: 'select' },
    },
    children: {
      name: 'children',
      description: 'The html wrapped inside the link',
      defaultValue: 'Test',
      control: { type: 'text' },
    },
    linkActiveStyle: {
      name: 'linkActiveStyle',
      description:
        'The styling to apply to the link and its contents when active',
      // defaultValue: css`
      //   color: #000000;
      // `,
      control: { type: 'object' },
    },
  },
};

export default meta;
type Story = StoryObj<typeof PageLink>;

export const Primary: Story = {
  args: {
    type: 'paintDrip',
    to: '/',
    direction: 'up',
    children: 'Test',
    linkActiveStyle: 'text-red-500!',
  },
};
