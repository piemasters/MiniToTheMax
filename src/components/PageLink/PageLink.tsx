import React, { FC } from 'react';
import { Link } from 'gatsby-link';

type AnimationTypes = 'paintDrip' | 'fade' | 'swipe' | 'cover';

export interface PageLinkProps {
  children: React.ReactNode;
  type?: AnimationTypes;
  to: string;
  linkActiveStyle?: string;
  direction?: string;
  disabled?: boolean;
  className?: string;
}

export const PageLink: FC<PageLinkProps> = ({
  children,
  type,
  to,
  linkActiveStyle,
  direction,
  disabled,
  className,
}) => {
  const baseClasses =
    className ||
    'text-blue-500 visited:text-blue-500 hover:text-blue-700 no-underline';

  if (disabled) {
    return <div className="text-gray-700">{children}</div>;
  }

  return (
    <Link
      data-testid="page-link"
      to={to}
      className={baseClasses}
      activeClassName={linkActiveStyle}
      partiallyActive={to !== '/'}
    >
      {children}
    </Link>
  );

  // if (type === 'paintDrip') {
  //   return (
  //     <AniLink
  //       data-testid="page-link-paint"
  //       paintDrip
  //       hex={hex ? hex : '#eb1d23'}
  //       to={to}
  //       duration={duration}
  //       css={activeStyle}
  //       activeStyle={linkActiveStyle}
  //       partiallyActive={to !== '/'}
  //     >
  //       {children}
  //     </AniLink>
  //   );
  // } else if (type === 'fade') {
  //   return (
  //     <AniLink
  //       data-testid="page-link-fade"
  //       fade={type === 'fade'}
  //       to={to}
  //       duration={duration}
  //       css={activeStyle}
  //       activeStyle={linkActiveStyle}
  //       partiallyActive={to !== '/'}
  //     >
  //       {children}
  //     </AniLink>
  //   );
  // } else if (type === 'swipe') {
  //   return (
  //     <AniLink
  //       data-testid="page-link-swipe"
  //       swipe
  //       direction={direction ? direction : 'up'}
  //       top={top ? top : 'exit'}
  //       entryOffset={entryOffset ? entryOffset : 100}
  //       hex={hex ? hex : '#eb1d23'}
  //       to={to}
  //       duration={duration}
  //       css={activeStyle}
  //       activeStyle={linkActiveStyle}
  //       partiallyActive={to !== '/'}
  //     >
  //       {children}
  //     </AniLink>
  //   );
  // } else if (type === 'cover') {
  //   return (
  //     <AniLink
  //       data-testid="page-link-cover"
  //       cover
  //       to={to}
  //       direction={direction ? direction : 'up'}
  //       bg={bg ? bg : '#eb1d23'}
  //       duration={duration ? duration : 1}
  //       css={activeStyle}
  //       activeStyle={linkActiveStyle}
  //       partiallyActive={to !== '/'}
  //     >
  //       {children}
  //     </AniLink>
  //   );
  // } else {
  //   return (
  //     <TransitionLink
  //       data-testid="page-link"
  //       to={to}
  //       css={activeStyle}
  //       activeStyle={linkActiveStyle}
  //       partiallyActive={to !== '/'}
  //     >
  //       {children}
  //     </TransitionLink>
  //   );
  // }
};

export default PageLink;
