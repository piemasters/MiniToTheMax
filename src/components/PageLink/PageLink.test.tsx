import { getByTestId, render } from '@testing-library/react';

import { PageLink } from './PageLink';

describe('PageLink', () => {
  test('renders correctly', () => {
    const { container } = render(<PageLink to={'/'}>Default</PageLink>);

    expect(
      getByTestId(container as HTMLElement, 'page-link')
    ).toHaveTextContent('Default');
    expect(container).toMatchSnapshot();
  });
});
