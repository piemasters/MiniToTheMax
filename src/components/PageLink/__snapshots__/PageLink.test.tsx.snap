// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLink renders correctly 1`] = `
<div>
  <a
    aria-current="page"
    class="text-blue-500 visited:text-blue-500 hover:text-blue-700 no-underline"
    data-testid="page-link"
    href="/"
  >
    Default
  </a>
</div>
`;
